Significant changes in the project

Dolgoff62 opened the pull request.

Dolgoff62 committed:
fix inconsistencies in names in layouts

Dolgoff62 committed:
All bugs and inconsistencies have been fixed. The project is being builded and fully operational.

Dolgoff62 committed:
Updated fun of transfer claim status from open to in_progress. Fix some layout errors

Dolgoff62 committed:
work on the method of processing an claim according to the status model

Dolgoff62 committed:
Completed work on status processing of claim.

nanstas committed:
change on include and main page

Dolgoff62 committed:
Completed work with status processing of claims.

Dolgoff62 committed:
try

Dolgoff62 committed:
added custom app bar

Dolgoff62 committed:
Update fragment_open_claim.xml

nanstas committed:
change ImageButton on MaterialButton

on:
container_list_claim.xml
container_list_news.xml

nanstas committed:
Merge branch 'road_to_first_release' of https://github.com/Comdcomp/FMHAndroid into road_to_first_release

Dolgoff62 committed:
The full functionality of navigation through the main menu has been implemented.

Dolgoff62 committed:
Changing the title and subtitle of several fragments

CreateEditClaim
CreateEditNews
CreateEditClaimComment

nanstas committed:
impl logic on MainFragment

Dolgoff62 committed:
update openClamFragment tool bar

Dolgoff62 committed:
Merge branch 'road_to_first_release' of https://github.com/Comdcomp/FMHAndroid into road_to_first_release

Dolgoff62 committed:
little changes

nanstas committed:
Update fragment_main.xml

Dolgoff62 committed:
change user id

Dolgoff62 committed:
Merge branch 'road_to_first_release' of https://github.com/Comdcomp/FMHAndroid into road_to_first_release

Dolgoff62 committed:
change warning displaying from snckbar to Toast

Dolgoff62 committed:
Fixed collision of two views on each other.

Dolgoff62 committed:
changed application icon

Dolgoff62 committed:
try to fix saveNews Exception warning

Dolgoff62 committed:
Continue fix save news bug

Dolgoff62 committed:
bugs fix

Dolgoff62 committed:
information button deleted from all fragment

nanstas committed:
add fragment_auth.xml

nanstas committed:
add splash screen

Dolgoff62 committed:
fix bug in news control panel

Dolgoff62 committed:
bug fix

fix some bugs with news list filtrating.

Dolgoff62 committed:
an attempt to solve the problem with displaying correct information on the claim after the repeated status change.

nanstas committed:
add few images on splash screen

nanstas committed:
add change background for spalsh screen title

Dolgoff62 committed:
work with open claim fragment

Dolgoff62 committed:
little changes

nanstas committed:
changed icons

Dolgoff62 committed:
Fixed status processing of claim. Fixed a bug with incorrect display of information in the claim. Minor changes.

Dolgoff62 committed:
Merge remote-tracking branch 'origin/road_to_first_release' into road_to_first_release

nanstas committed:
some layout changes

nanstas committed:
add active title for toggle in CreateEditNews

nanstas committed:
some layout changes

Dolgoff62 committed:
Improvement in openClaimFragment

Dolgoff62 committed:
The status processing of claims has been completely improved

Dolgoff62 committed:
Merge branch 'work_with_Maksim' into road_to_first_release

Dolgoff62 committed:
Update ClaimViewModel.kt

Dolgoff62 committed:
bug fixes

Dolgoff62 committed:
increased area of pressing the main menu button

Dolgoff62 committed:
minor code fixes

Dolgoff62 committed:
Bugs fixed in claims and claim comments

Dolgoff62 committed:
update kotlin version in build gradle

Dolgoff62 committed:
minor changes

Dolgoff62 committed:
layout background added to main fragment

Dolgoff62 committed:
set backgroud to main fragment layout

nanstas committed:
change images for splashscreen

Dolgoff62 committed:
dependency library updated, also updated app background

Dolgoff62 committed:
Update claim and news shared flows

nanstas committed:
Merge branch 'road_to_first_release' of https://github.com/fmh-charity/fmh-android into road_to_first_release

 Conflicts:
	app/src/main/res/layout/activity_splash_screen.xml

Dolgoff62 committed:
implemented swipe to refresh to main fragment

Dolgoff62 committed:
Merge remote-tracking branch 'origin/road_to_first_release' into road_to_first_release

Dolgoff62 committed:
added app_background image in 1920x1920 resolution

Dolgoff62 committed:
fix bug with swipe to refresh in main fragment

Dolgoff62 committed:
bug fixed in claim list fragment

Fixed a bug when, when adding a new item to the list, scrolling to the beginning of the new list did not occur.

Dolgoff62 committed:
updated okhttploggin_version to "4.9.2"

Dolgoff62 committed:
added swipe to refresh to news list fragment

Dolgoff62 committed:
refactor events in claimCardViewModel

Dolgoff62 committed:
minor changes

Dolgoff62 committed:
finall refactoring with Events

All Events removed from claimCardViewModel.

Dolgoff62 committed:
Update item_claim.xml

Dolgoff62 committed:
Major changes in some classes

Final refactoring of Events throughout the project.
Updated our mission button icon. Added a method for checking the availability of the Internet on a smartphone. In the corresponding fragment, the indents between the claim cards have been increased.

Dolgoff62 committed:
Some changes

fix our mission icon. upgrade build gradle dependency version.
Start to refactor flow collect to repeatOnLifecycle

Dolgoff62 committed:
Revert to old indents between claim cards.

Dolgoff62 committed:
added swipe-to-refresh to fragment news control panel

Dolgoff62 committed:
Merge branch 'refactoring_events' into road_to_first_release

Dolgoff62 committed:
Merge branch 'main' into road_to_first_release

Dolgoff62 committed:
fix bug with white spaces

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+    fun onCard(claimComment: ClaimCommentWithCreator) {}
+}
+
+// Временная переменная. После авторизации заменить на залогиненного юзера
+val user = User(
+    id = 1,
+    login = "User-1",
+    password = "abcd",
+    firstName = "Дмитрий",
+    lastName = "Винокуров",
+    middleName = "Владимирович",
+    phoneNumber = "+79109008765",
+    email = "Vinokurov@mail.ru",
+    deleted = false
+)
```
Спрячь глубже, чтобы не было мучительно больно при авторизации прятать. Сделай, как это было бы на самом деле. Если я всё правильно понимаю, то у тебя этот заглушечный пользователь должен всплыть где-то на уровне имплементации репозитория. Там же он и умрёт, когда появится авторизация

Dolgoff62 replied:
До авторизации оставлю. Чтобы не делать двойную работу

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    override fun onBindViewHolder(holder: ClaimCommentViewHolder, position: Int) {
+        getItem(position)?.let {
```
Лучше без `?`. Элемент всегда должен быть. Пускай в таком случае приложение лучше упадёт, если `getItem(position)` вернёт `null`

Dolgoff62 replied:
исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+                commentatorNameTextView.text = claimComment.creator.lastName?.let { lastName ->
+                    claimComment.creator.firstName?.let { firstName ->
+                        claimComment.creator.middleName?.let { middleName ->
+                            Utils.shortUserNameGenerator(
```
```suggestion
                            Utils.generateShortUserName(
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+                commentDescriptionTextView.text = claimComment.claimComment.description
+                commentatorNameTextView.text = claimComment.creator.lastName?.let { lastName ->
+                    claimComment.creator.firstName?.let { firstName ->
+                        claimComment.creator.middleName?.let { middleName ->
```
классический `if` и SmartCast сделают всё понятнее и без лишних вложенностей :wink:

nanstas replied:
Избавились от null, вложенность исчезла)

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+                    }
+                }
+                commentDateTextView.text =
+                    claimComment.claimComment.createDate?.let {Utils.showDate(it)}
```
Неудачное название метода :man_shrugging: 
Как будто дата должна быть где-то показана, например, в консоли. Предложу вариант, чтобы было понятнее о чём я.
```suggestion
                    claimComment.claimComment.createDate?.let { Utils.formatDate(it) }
```

maks1596 replied:
то же самое и с `showTime`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+                    editLightImageButton.setImageResource(R.drawable.ic_pen_light)
+                } else {
+                    editLightImageButton.setImageResource(R.drawable.ic_pen)
+                }
```
По-моему так короче будет и не менее выразительно :blush:  
```suggestion
                editLightImageButton.setImageResource(
                    if (claimComment.creator.id != user.id) R.drawable.ic_pen_light else R.drawable.ic_pen
                )
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+        }
+    }
+
+    class ClaimCommentDiffCallback : DiffUtil.ItemCallback<ClaimCommentWithCreator>() {
```
```suggestion
    prvate object DiffCallback : DiffUtil.ItemCallback<ClaimCommentWithCreator>() {
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+            newItem: ClaimCommentWithCreator
+        ): Boolean {
+            return oldItem == newItem
+        }
```
Реализация неверная. Читать документацию к методам

nanstas replied:
Исправлено

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,106 @@
+    ) : RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(claimComment: ClaimCommentWithCreator) {
+            binding.apply {
```
На случай, если захочется поэкономить место и не вызвать побочных эффектов
```suggestion
        fun bind(claimComment: ClaimCommentWithCreator) = with(binding) {
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimListAdapter.kt:
```diff
@@ -0,0 +1,93 @@
+                    executorNameMaterialTextView.setText(R.string.not_assigned)
+                } else {
+                    executorNameMaterialTextView.text = Utils.shortUserNameGenerator(
+                        fullClaim.executor?.lastName.toString(),
```
зачем `toString` в конце? :astonished: 
чтобы от `nullability` избавиться?

maks1596 replied:
Если что, то имплементация этого `toString` примерно следующая:
```kotlin
fun String?.toString() = this ?: "this string is null"
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimListAdapter.kt:
```diff
@@ -0,0 +1,93 @@
+                            it
+                        )
+                    }
+                themeMaterialTextView.text = fullClaim.claim.title
```
что-то название какое-то у `TextView` странное :thinking:

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/NewsControlPanelAdapter.kt:
```diff
@@ -0,0 +1,121 @@
+                        itemView.context.getString(R.string.news_control_panel_not_active)
+                    newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_clear_24)
+                }
+            }
```
Серьёзно? `if ... else` похоже теперь стал "для слабаков" :smile: 
```suggestion
            if (newsItemWithCreator.news.newsItem.publishEnabled) {
                newsItemPublishedTextView.text = itemView.context.getString(R.string.news_control_panel_active)
                newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_check_24)
            } else {
                newsItemPublishedTextView.text = itemView.context.getString(R.string.news_control_panel_not_active)
                newsItemPublishedIconImageView.setImageResource(R.drawable.ic_baseline_clear_24)
            }
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/NewsControlPanelAdapter.kt:
```diff
@@ -0,0 +1,121 @@
+                    6 -> R.raw.icon_massage
+                    7 -> R.raw.icon_gratitude
+                    8 -> R.raw.icon_help
+                    else -> return
```
никаких "1, 2, 3..." и так далее. Переделать на `enum` или что-то около того

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/NewsControlPanelAdapter.kt:
```diff
@@ -0,0 +1,121 @@
+                newsItemWithCreator.user.lastName,
+                newsItemWithCreator.user.firstName?.first()?.plus("."),
+                newsItemWithCreator.user.middleName?.first()?.plus(".")
+            )
```
По-моему для этого есть метод в `Utils`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/NewsListAdapter.kt:
```diff
@@ -0,0 +1,85 @@
+
+class NewsDiffCallBack : DiffUtil.ItemCallback<NewsWithCreators>() {
+    override fun areItemsTheSame(oldItem: NewsWithCreators, newItem: NewsWithCreators): Boolean {
+        return oldItem == newItem
```
Неправильная имплементация

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/api/ApiModule.kt:
```diff
@@ -0,0 +1,32 @@
+    @Provides
+    @Singleton
+    fun provideClaimApi(): ClaimApi {
+        return retrofit(okhttp(loggingInterceptor()))
```
`loggingInterceptor()` похоже на какую-то глобальную функцию. От куда она?
Может лучше будет этот `interceptor` заинжектить в метод? :man_shrugging:

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/application/FhmApplication.kt:
```diff
@@ -0,0 +1,11 @@
+class FhmApplication: Application() {
+    override fun onCreate() {
+        super.onCreate()
+    }
```
В `onCreate` не ругается на бесполезное переопределение метода?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,54 @@
+
+@Dao
+interface ClaimDao {
+    // Сортируй тут по датам дурачок :)
```
М?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,54 @@
+    fun getClaimsOpenAndInProgressStatuses(
+        firstStatus: Claim.Status,
+        secondStatus: Claim.Status
+    ): Flow<List<FullClaim>>
```
Так не универсальнее будет?
```suggestion
    @Query("SELECT * FROM ClaimEntity WHERE status IN :statuses ORDER BY planExecuteDate ASC, createDate DESC")
    fun getClaimsInStatuses(statuses: List<Claim.Status>): Flow<List<FullClaim>>
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,54 @@
+    ): Flow<List<FullClaim>>
+
+    @Query("SELECT * FROM ClaimCommentEntity WHERE claimId = :claimId")
+    fun getClaimComments(claimId: Int): Flow<List<ClaimCommentWithCreator>>
```
К другой таблице обращаешься. Я бы в отдельный `DAO` убрал, если честно

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,54 @@
+    suspend fun deleteClaimById(id: Int)
+}
+
+class WishClaimStatusConverter {
```
Почему `Wish`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/db/AppDb.kt:
```diff
@@ -1,42 +1,29 @@
+    abstract fun userDao(): UserDao
+    abstract fun claimDao(): ClaimDao
+    abstract fun newsDao(): NewsDao
+    abstract fun newsCategoryDao(): NewsCategoryDao
```
Обычно так методы не именуются. Либо поменяй `fun` на `val`, либо добавь в начале всех методов `get`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Claim.kt:
```diff
@@ -0,0 +1,58 @@
+    val planExecuteDate: Long? = null,
+    val factExecuteDate: Long? = null,
+    var status: Status? = null,
+    val deleted: Boolean = false,
```
Что-то мне все эти значения по умолчанию не нравятся. Можно без них обойтись?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/Claim.kt:
```diff
@@ -0,0 +1,58 @@
+    )
+    val comments: List<ClaimCommentWithCreator>?
+
+) : Parcelable
```
Убери в отдельный файл. `Claim` (как мне мерещится) - это чистая модель данных, а `FullClaim` - это немного более огрязнённая деталями вытяжка из базы данных

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/News.kt:
```diff
@@ -0,0 +1,61 @@
+    ) : Parcelable {
+        override fun toString(): String {
+            return name
+        }
```
чёт падазрительна :raised_eyebrow: 
Выбери либо `toString`, либо `data class`. Вместе это немного странно выглядит

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/SplashScreen.kt:
```diff
@@ -0,0 +1,8 @@
+package ru.netology.fmhandroid.dto
+
+data class SplashScreen(
```
Странное какое-то название для dto

maks1596 replied:
- `SplashScreenData`
- `SplashScreenConfiguration`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,23 @@
+
+interface ClaimRepository {
+    val data: Flow<List<FullClaim>>
+    val dataOpenInProgress: Flow<List<FullClaim>>
```
Для единообразия лучше их тоже превратить в методы

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,23 @@
+interface ClaimRepository {
+    val data: Flow<List<FullClaim>>
+    val dataOpenInProgress: Flow<List<FullClaim>>
+    suspend fun getAllClaims(): List<Claim>
```
Непонятно чем этот метод так принципиально отличается от `data`. Ну то есть:
1. Почему `suspend`?
2. Почему возвращает `List`, а не `Flow`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,23 @@
+    suspend fun changeClaimStatus(
+        claimId: Int,
+        newStatus: Claim.Status,
+        executorId: Int?,
+        claimComment: ClaimComment
```
Без комментария мне не понятно, как в смене статуса участвуют последние два аргумента

maks1596 replied:
что-то мне подсказывает, что надо делать 4 отдельных метода с собственным специфичным набором аргументов, а не пытаться всё свести под одну гребёнку

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,23 @@
+    val dataOpenInProgress: Flow<List<FullClaim>>
+    suspend fun getAllClaims(): List<Claim>
+    suspend fun editClaim(editedClaim: Claim): Claim
+    suspend fun saveClaim(claim: Claim): Claim
```
Затрудняюсь вот так сходу в интерфейсе понять чем отличаются эти два метода друг от друга?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,23 @@
+        claimComment: ClaimComment
+    ): Claim
+    suspend fun changeClaimComment(comment: ClaimComment): ClaimComment
+    suspend fun getAllClaimsWithOpenAndInProgressStatus(): List<Claim>
```
А в чём отличие от `dataOpenInProgress`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/newsRepository/NewsRepositoryImpl.kt:
```diff
@@ -0,0 +1,145 @@
+
+    override suspend fun getAllNewsCategories(): Flow<List<News.Category>> =
+        newsCategoryDao.getAllNewsCategories().map { it.toNewsCategoryDto() }
+            .catch { e -> from(e) }
```
Воу-воу. Это что за `from` и зачем он?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/newsRepository/NewsRepositoryImpl.kt:
```diff
@@ -0,0 +1,145 @@
+
+    // Метод подлежит удалению после реализации добавления новых категорий
+    override suspend fun saveCategories() =
+        newsCategoryDao.insert(categories.toNewsCategoryEntity())
```
Может всю это грязнульку перенести в `App`? Там при старте приложения проверить что всё, что нам нужно есть, а если нет, то заинсёртить

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/userRepository/UserRepositoryImp.kt:
```diff
@@ -1,29 +1,74 @@
+            .flowOn(Dispatchers.Default)
+
+    override suspend fun getAllUsers() {
+        val temporaryUserList = mutableListOf(User(
```
Просто `listOf` уже не в моде или не подходит?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/userRepository/UserRepositoryImp.kt:
```diff
@@ -1,29 +1,74 @@
+                deleted = false
+            )
+        )
+        userDao.insert(temporaryUserList.toEntity())
```
На каждый вызов метода будет происходить вставка пользователей в таблицу. Зачем? Может проще при старте приложения проверить пустая ли таблица с пользователями, и если да, то заполнить таблицу

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/AppActivity.kt:
```diff
@@ -1,12 +1,21 @@
+        // Костыль!!! Убрать!!! (Only for test!)
+        val viewModel: UserViewModel by viewModels()
+
+        val bla = viewModel
```
Настало время убрать? :blush:

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+
+        binding = FragmentListClaimBinding.bind(view)
+
+        val menuFiltering =
```
```suggestion
        val filteringMenu =
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+            binding.containerCustomAppBarIncludeOnFragmentListClaim.mainMenuImageButton
+        )
+        mainMenu.inflate(R.menu.menu_main)
+        mainMenu.menu.removeItem(R.id.menu_item_claims)
```
Я не понимаю зачем сразу удалять элемент из меню

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+        mainMenu.setOnMenuItemClickListener {
+            when (it.itemId) {
+                R.id.menu_item_main -> {
+                    findNavController().navigate(R.id.action_claimListFragment_to_mainFragment)
```
Лучше прокинуть во `viewModel`, а она уже решит что делать с этим нажатием по пункту меню

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+        }
+
+        binding.apply {
+            containerListClaimInclude.expandMaterialButton.visibility = View.GONE
```
Если ничего не мешает то проще в файл с разметкой прописать
```xml
android:visibility="gone"
tools:visibility="visible"
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+        binding.apply {
+            containerListClaimInclude.expandMaterialButton.visibility = View.GONE
+            containerListClaimInclude.allClaimsTextView.visibility = View.GONE
+            root.setBackgroundResource(R.drawable.background_app)
```
Что-то мешает установить задник в разметке?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+                        }
+                    }
+                }
+            }
```
Обработка события нажатия по карточке должна быть во `viewModel`-и

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFragment.kt:
```diff
@@ -0,0 +1,216 @@
+        R.id.executes_list_item -> {
+            lifecycleScope.launchWhenCreated {
+                viewModel.data.collectLatest { state ->
+                    adapter.submitList(state.filter { it.claim.status == Claim.Status.EXECUTED })
```
У тебя есть база. Фильтрация там эффективнее произойдёт

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditClaimCommentFragment.kt:
```diff
@@ -0,0 +1,139 @@
+                showErrorToast(R.string.error)
+            }
+        }
+        lifecycleScope.launch {
+            claimCardViewModel.claimCommentCreatedEvent.collect {
+                findNavController().navigateUp()
+            }
+        }
+        lifecycleScope.launch {
+            claimCardViewModel.claimCommentUpdatedEvent.collect {
+                findNavController().navigateUp()
+            }
+        }
```
`View` не должен знать как реагировать на события. `View` - не делегат. `View` глупая - что ей скажут, то и сделает. Поэтому эту пару событий совершенно точно можно схлопнуть до одного прямого указания "навигируемся назад", "закрываем экран". Ну, или ещё чего, на что фантазии хватит :blush:

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditClaimCommentFragment.kt:
```diff
@@ -0,0 +1,139 @@
+        val binding = FragmentCreateEditCommentBinding.bind(view)
+
+        val args: CreateEditClaimCommentFragmentArgs by navArgs()
+        val comment: ClaimCommentWithCreator? = args.argComment
+        val claimId: Int = args.argClaimId
```
Надо все аргументы отдать во `viewModel`. `viewModel` над ними посидит, подумает, запросит нужные данные, отбросит всю шелуху, и когда данные будут готовы для показа, закинет их во `Flow`, на который уже подписан `Fragment`.
Либо совсем не использовать `ViewModel` на этом экране, если он слишком прост, чтобы его усложнять слишком правильной архитектурой

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditClaimCommentFragment.kt:
```diff
@@ -0,0 +1,139 @@
+        val comment: ClaimCommentWithCreator? = args.argComment
+        val claimId: Int = args.argClaimId
+
+        with(binding) {
+            containerCustomAppBarIncludeOnFragmentCreateEditClaimComment.mainMenuImageButton.visibility =
+                View.GONE
+            containerCustomAppBarIncludeOnFragmentCreateEditClaimComment.authorizationImageButton.visibility =
+                View.GONE
+            containerCustomAppBarIncludeOnFragmentCreateEditClaimComment.ourMissionImageButton.visibility =
+                View.GONE
+        }
```
```suggestion
        with(binding.containerCustomAppBarIncludeOnFragmentCreateEditClaimComment) {
            mainMenuImageButton.visibility = View.GONE
            authorizationImageButton.visibility = View.GONE
            ourMissionImageButton.visibility = View.GONE
        }
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditClaimCommentFragment.kt:
```diff
@@ -0,0 +1,139 @@
+                .setText(R.string.genitive_comment)
+
+            binding.saveButton.setOnClickListener {
+                // TODO("Доработать когда будет реализована авторизация
+                //  Вставлять корректный creatorId вместо хардкода")
```
Спрячь всё это дальше в приложении. Ближе к данным. Ближе к тем местам, где реально надо будет что-то менять.

P.S.: В TODO-_**комментарии**_ совсем не обязательно ставить круглые скобки и кавычки, как при вызове _TODO-**функции**_ :hand_over_mouth:

maks1596 reviewed (changes requested):
Комментарии не дублировал. Для каждого комментария необходимо искать аналогичные места и исправлять их тоже

Dolgoff62 committed:
Fix pull request issues with claim and news

Dolgoff62 committed:
start to refactor list of claim

Dolgoff62 committed:
claim refactoring

Dolgoff62 committed:
Improvements to claim list fragment

Dolgoff62 committed:
improved claim list fragment. Work started on fixes in the pull request

Dolgoff62 committed:
refactoring in all adapters

Dolgoff62 committed:
remaster dialog in claimListFragment

Dolgoff62 committed:
refactoring news category, and some minor changes

Dolgoff62 committed:
Added claimCommentDao, create FullClaim data class

Dolgoff62 committed:
refactor di, created localDateTime to timeStamp converter

Dolgoff62 committed:
start try

Dolgoff62 committed:
second try

Dolgoff62 committed:
final fix problem with news list fragment

Dolgoff62 committed:
Merge branch 'try_to_refactor_filter_news' into road_to_first_release

Dolgoff62 committed:
created ClaimListFilteringDialogFragment

refactor clain list filtering

Dolgoff62 committed:
added examination current destination with navigation

this added in claimListFragment and mainFragment

Dolgoff62 committed:
fix bug with claims

Dolgoff62 committed:
minor changes in some classes

Dolgoff62 committed:
refactor some button size

Dolgoff62 committed:
major refactor

Dolgoff62 committed:
bug fixes

bug with deleted id's fixed. refactor newsViewModel data Flow

Dolgoff62 committed:
Merge branch 'latest_changes_not_ended' into road_to_first_release

Dolgoff62 committed:
delete unusual function in some classes

Dolgoff62 committed:
refactor in dto and viewModels

Dolgoff62 committed:
work with issue in pull request

Dolgoff62 committed:
work eith issue in pull request

Dolgoff62 committed:
Merge remote-tracking branch 'origin/default_null_properties_refactoring' into default_null_properties_refactoring

nanstas committed:
renaming methods

Dolgoff62 committed:
Moving the list of news categories to AppActivity

Dolgoff62 committed:
clean up code in multiple classes

Dolgoff62 committed:
fix issue about claimListFragment background

nanstas committed:
binding refactoring

Dolgoff62 committed:
method refactor

Dolgoff62 committed:
Added functionality for processing status with empty lists of news and claims

Dolgoff62 committed:
The overwhelming majority of comments have been corrected after the code has been inspected for the entire project.

nanstas committed:
add ScrollView

Dolgoff62 committed:
minor changes

Dolgoff62 committed:
fix bug with scroll to position

Dolgoff62 committed:
Added the ability to copy the description of the claim, comments and news into the clipboard

Dolgoff62 committed:
claim comment item background changed

nanstas committed:
fix bag with open & close item in recyclerView

Dolgoff62 committed:
Editing the background for comments

Dolgoff62 committed:
icon for authorization and main menu button changed

nanstas committed:
fix bug with open & close item in recyclerView (on onNewsControlPanelFragment)

Dolgoff62 committed:
Major refactoring with open and close news item

Dolgoff62 committed:
refactor dto

maks1596 commented on app/lint.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Хочу узнать всю предысторию появления этого файлика 😊

Dolgoff62 committed:
refactor auth api module and his chain

maks1596 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,35 +1,42 @@
+        android:usesCleartextTraffic="true"
+        android:theme="@style/Theme.FMHAndroid"
+        android:name=".application.FhmApplication"
+        tools:targetApi="m">
```
К чему эта целевая версия приписана?

maks1596 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,35 +1,42 @@
+
+        <activity android:name=".ui.SplashScreenActivity"
+            android:screenOrientation="portrait"
+            android:exported="true"
```
не надо экспортировать стартовую активити. и так должно работать

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/ClaimCommentListAdapter.kt:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    override fun onBindViewHolder(holder: ClaimCommentViewHolder, position: Int) {
+        getItem(position).let {
```
лучше не надо использовать `let` только для короткого имени `it`. `let` полезен с `nullable` значениями: `nullableVariable?.let { ... }`. в остальных случаях `let` не несёт в себе полезной нагрузки, и более читаемым вариантом будет просто положить значение в переменную с коротким именем (`it`, если так угодно)

nanstas replied:
положил в переменную.

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/adapter/NewsControlPanelAdapter.kt:
```diff
@@ -0,0 +1,125 @@
+                    onInteractionListener.onCard(newsItemWithCreator.news.newsItem)
+                }
+
+                when (newsItemWithCreator.news.newsItem.publishEnabled) {
```
Заменить на `if else`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimCommentDao.kt:
```diff
@@ -0,0 +1,21 @@
+    fun getClaimComments(claimId: Int): Flow<List<ClaimCommentWithCreator>>
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertComment(comments: List<ClaimCommentEntity>)
```
```suggestion
    suspend fun insertComments(comments: List<ClaimCommentEntity>)

```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,46 @@
+    """
+    )
+    fun getClaimsByStatus(
+        listStatuses: List<Claim.Status>
```
Неправильное название: либо `statusList`, либо `statuses`. `listStatuses` - это "статусы списка"

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/ClaimDao.kt:
```diff
@@ -0,0 +1,46 @@
+    suspend fun insertClaim(claim: ClaimEntity)
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    suspend fun insertClaim(claims: List<ClaimEntity>)
```
```suggestion
    suspend fun insertClaims(claims: List<ClaimEntity>)

```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dao/NewsDao.kt:
```diff
@@ -0,0 +1,53 @@
+}
+
+@Dao
+interface NewsCategoryDao {
```
Лучше бы, наверное, в отдельный файлик

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/db/AppDb.kt:
```diff
@@ -1,42 +1,30 @@
+    ClaimClaimStatusConverter::class
+)
+abstract class AppDb : RoomDatabase() {
+    abstract fun getUserDao(): UserDao
+    abstract fun getClaimDao(): ClaimDao
+    abstract fun getClaimCommentDao(): ClaimCommentDao
+    abstract fun getNewsDao(): NewsDao
+    abstract fun getNewsCategoryDao(): NewsCategoryDao
```
Если вдруг что, то можно и на свойства поменять 😉  
Например, `abstract val userDao: UserDao`. Так немного сэкономим на круглых скобках `()`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepository.kt:
```diff
@@ -0,0 +1,31 @@
+     * При переводе заявки из статуса OPEN -> IN PROGRESS заявке назначается исполнитель
+     * При переводе заявки из статуса IN PROGRESS -> OPEN с заявки снимается исполнитель
+     * При действиях "Сбросить" и "Исполнить" исполнитель обязан оставить объясняющий комментарий
+     * **/
```
Спасибо за комментарий. Теперь лучше видно, что надо делать 4 отдельных метода с собственным специфичным набором параметров, а не пытаться всё свести под одну гребёнку 😊

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/dto/SplashScreen.kt:
```diff
@@ -0,0 +1,11 @@
+
+@SuppressLint("CustomSplashScreen")
+data class SplashScreen(
+    val image: Int,
```
Почему изображение - целое число?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/claimRepository/ClaimRepositoryImpl.kt:
```diff
@@ -0,0 +1,111 @@
+
+    override fun getClaimsByStatus(
+        coroutineScope: CoroutineScope,
+        listStatuses: List<Claim.Status>
```
либо `statuses`, либо `statusList`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/newsRepository/NewsRepository.kt:
```diff
@@ -0,0 +1,24 @@
+
+interface NewsRepository {
+    suspend fun refreshNews()
+    suspend fun modificationOfExistingNews(newsItem: News): News
```
`update` что ли? 🤔

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/newsRepository/NewsRepository.kt:
```diff
@@ -0,0 +1,24 @@
+        dateStart: Long? = null,
+        dateEnd: Long? = null
+    ): Flow<List<NewsWithCreators>>
+    suspend fun changeIsOpen(newsItem: News)
```
можно комментарий? или метод переименовать? непонятное название

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/repository/newsRepository/NewsRepositoryImpl.kt:
```diff
@@ -0,0 +1,83 @@
+
+    override suspend fun changeIsOpen(newsItem: News) {
+        newsDao.insert(newsItem.toEntity())
+    }
```
Может метод тогда `update` назвать или `save`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/AppActivity.kt:
```diff
@@ -1,12 +1,38 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_app)
+
+        // Костыль!!! Убрать!!! (Only for test!)
```
```suggestion
        // TODO Костыль!!! Убрать!!! (Only for test!)
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/AppActivity.kt:
```diff
@@ -1,12 +1,38 @@
+                News.Category(6, "Массаж", false),
+                News.Category(7, "Благодарность", false),
+                News.Category(8, "Нужна помощь", false)
+            )
```
По-моему я уже в репозитории видел этот список категорий, разве нет?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFilteringDialogFragment.kt:
```diff
@@ -0,0 +1,92 @@
+    }
+
+
+    private fun mutableListOfClaimStatus(dialogView: View): MutableList<Claim.Status> {
```
```suggestion
    private fun View.getCheckedStatuses(): List<Claim.Status> {
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/ClaimListFilteringDialogFragment.kt:
```diff
@@ -0,0 +1,92 @@
+        if (dialogView.findViewById<MaterialCheckBox>(R.id.item_filter_executed).isChecked)
+            checkedStatusList.add(Claim.Status.EXECUTED)
+        if (dialogView.findViewById<MaterialCheckBox>(R.id.item_filter_cancelled).isChecked)
+            checkedStatusList.add(Claim.Status.CANCELLED)
```
```suggestion
        fun check(checkBoxId: Int, status: Claim.Status) {
            if (dialogView.findViewById<MaterialCheckBox>(checkBoxId).isChecked)
                 checkedStatusList.add(status)
        }
        check(R.id.item_filter_open, Claim.Status.OPEN)
        check(R.id.item_filter_in_progress, Claim.Status.IN_PROGRESS)
        check(R.id.item_filter_executed, Claim.Status.EXECUTED)
        check(R.id.item_filter_cancelled, Claim.Status.CANCELLED)
```

maks1596 replied:
Или что-нибудь подобное, чтобы компилятор обругивал на случай, если забудешь отразить в фильтре новый статус  
```suggestion
        for (status in Claim.Status.values()) {
             fun check(checkBoxId: Int) {
                 if (dialogView.findViewById<MaterialCheckBox>(checkBoxId).isChecked)
                     checkedStatusList.add(status)
             }
             when(status) {
                 Claim.Status.CANCELLED -> check(R.id.item_filter_cancelled)
                 Claim.Status.EXECUTED -> check(R.id.item_filter_executed)
                 Claim.Status.IN_PROGRESS -> check(R.id.item_filter_in_progress)
                 Claim.Status.OPEN -> check(R.id.item_filter_open)
             }
        }
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditClaimFragment.kt:
```diff
@@ -0,0 +1,243 @@
+    private lateinit var vTimePicker: TextInputEditText
+    private lateinit var binding: FragmentCreateEditClaimBinding
+    private val args: CreateEditClaimFragmentArgs by navArgs()
+    //временно, пока нет авторизации
```
```suggestion
    // TODO временно, пока нет авторизации
```

maks1596 commented on app/src/main/res/menu/menu_main.xml:
```diff
@@ -0,0 +1,18 @@
+        android:visible="false"/>
+    <item
+        android:title="@string/duty_roster"
+        android:visible="false"/>
```
Зачем эти пункты меню? 
1. без идентификаторов
2. невидимые

maks1596 commented on app/src/main/res/drawable/separator.xml:
```diff
@@ -1,19 +1,18 @@
+    android:height="1dp"
+    android:viewportWidth="311"
+    android:viewportHeight="1"
+    tools:ignore="VectorRaster">
```
От чего этот `ignore`?

maks1596 commented on app/src/main/res/drawable/ic_rectangle_381.xml:
```diff
@@ -0,0 +1,9 @@
+    android:viewportHeight="25">
+  <path
+      android:pathData="M6,1L71,1A5,5 0,0 1,76 6L76,19A5,5 0,0 1,71 24L6,24A5,5 0,0 1,1 19L1,6A5,5 0,0 1,6 1z"
+      android:fillColor="@android:color/transparent"/>
```
Прозрачный? Зачем нужен прозрачный `drawable`?

maks1596 commented on app/src/main/res/drawable/ic_our_mission_21x23.xml:
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="22dp"
+    android:height="24dp"
+    android:viewportWidth="21"
+    android:viewportHeight="23">
```
так какой размер-то всё-таки? зачем значения расходятся? 😊

maks1596 commented on app/src/main/res/drawable/background_splash_screen_title_1.xml:
```diff
@@ -1,6 +1,6 @@
-    android:shape="rectangle">
-    <solid android:color="@color/white" />
-    <corners android:radius="5dp" />
+<?xml version="1.0" encoding="utf-8"?>
```
6 файлов, которые отличаются только цветом. По сути тебе от фигуры внутри нужно только закругление в 5dp. Цвет можно попробовать наложиться через параметр `backgroundTint`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/viewmodel/UserViewModel.kt:
```diff
@@ -1,23 +1,24 @@
+
+    init {
+        viewModelScope.launch {
+            userRepository.getAllUsers()
```
похоже на `refresh`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/viewmodel/NewsViewModel.kt:
```diff
@@ -0,0 +1,135 @@
+            ASC -> DESC
+            DESC -> ASC
+        }
+    }
```
Перенеси это направление сортировки отдельно. Пригодится ещё ни раз

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+            "Нужна помощь" -> 8
+            else -> 0
+        }
+    }
```
?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+            creatorId = 0,
+            createDate = 0,
+        )
+    }
```
Где-нибудь используется? Может заменить на глобальную функцию `internal fun EmptyClaimComment()`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+        val format = "dd.MM.yyyy"
+        val simpleDateFormat = SimpleDateFormat(format, Locale.getDefault())
+        editText.setText(simpleDateFormat.format(calendar.time))
+    }
```
А если вынести из `Utils` и превратить в расширение для `EditText`?

```suggestion
internal fun EditText.setDate(calendar: Calendar) {
    val format = "dd.MM.yyyy"
    val simpleDateFormat = SimpleDateFormat(format, Locale.getDefault())
    setText(simpleDateFormat.format(calendar.time))
}
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+        val format = "HH:mm"
+        val simpleDateFormat = SimpleDateFormat(format, Locale.getDefault())
+        editText.setText(simpleDateFormat.format(calendar.time))
+    }
```
А если вынести из `Utils` и превратить в расширение для `EditText`?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+        val localTime = LocalTime.parse(time, timeFormatter)
+        val localDateTime = LocalDateTime.of(localDate, localTime)
+        return fromLocalDateTimeToTimeStamp(localDateTime)
+    }
```
Ничего этот метод не сохраняет. Не надо никого обманывать. Переименовать

maks1596 replied:
Комментарий, кстати, тоже не в тему: нет в методе никакого знания про `picker`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+            ZoneId.systemDefault()
+        )
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(
+            "dd.MM.yyy",
```
Выше был формат `dd.MM.yyyy`. Ничего не перепутано?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+                        hasTransport(NetworkCapabilities.TRANSPORT_VPN) -> {
+                            result = true
+                        }
+                    }
```
А так случайно не работает?
```suggestion
                    result = hasTransport(NetworkCapabilities.TRANSPORT_WIFI or NetworkCapabilities.TRANSPORT_CELLULAR or NetworkCapabilities.TRANSPORT_VPN)
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/utils/Utils.kt:
```diff
@@ -0,0 +1,166 @@
+        var result = false // Returns connection. false: none; true: mobile data or wifi
+        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            cm?.run {
```
Вынеси проверку на `null` перед `if else` - она у тебя дважды происходит

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/SplashScreenActivity.kt:
```diff
@@ -0,0 +1,68 @@
+@SuppressLint("CustomSplashScreen")
+@Suppress("DEPRECATION")
+class SplashScreenActivity : AppCompatActivity() {
+    private val splashscreenImages = listOf(
```
Уже не просто `images` 😊

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/SplashScreenActivity.kt:
```diff
@@ -0,0 +1,68 @@
+        SplashScreen(R.drawable.image_splashscreen_16, "Ответственно и осознанно нести добро людям", R.drawable.background_splash_screen_title_3, R.color.splash_screen_title_color_3),
+        SplashScreen(R.drawable.image_splashscreen_17, "Хоспис – это компетентная помощь и любовь к пациентам", R.drawable.background_splash_screen_title_4, R.color.splash_screen_title_color_4)
+    )
+    private val splashscreenImage = splashscreenImages.random()
```
что-то как-то расточительно хранить целый список ради одного случайного элемента, не находишь?
```suggestion
    private val splashscreenImage by lazy { 
    listOf(
    SplashScreen(R.drawable.image_splashscreen_1,"Любая помощь важна и нужна", R.drawable.background_splash_screen_title_1, R.color.splash_screen_title_color_1),
        SplashScreen(R.drawable.image_splashscreen_2, "Бережное отношение к пациентам и их близким", R.drawable.background_splash_screen_title_2, R.color.splash_screen_title_color_2),
        SplashScreen(R.drawable.image_splashscreen_3, "Творческий и осознанный подход к жизни до конца", R.drawable.background_splash_screen_title_2, R.color.splash_screen_title_color_2),
        SplashScreen(R.drawable.image_splashscreen_4, "Ответственно и осознанно нести добро людям", R.drawable.background_splash_screen_title_4, R.color.splash_screen_title_color_4),
        SplashScreen(R.drawable.image_splashscreen_5, "Помощь – это создание комфорта для пациентов и их близких", R.drawable.background_splash_screen_title_5, R.color.splash_screen_title_color_5),
        SplashScreen(R.drawable.image_splashscreen_6, "Ответственно и осознанно нести добро людям", R.drawable.background_splash_screen_title_1, R.color.splash_screen_title_color_1),
        SplashScreen(R.drawable.image_splashscreen_7, "Творческий и осознанный подход к жизни пациента", R.drawable.background_splash_screen_title_6, R.color.splash_screen_title_color_6),
        SplashScreen(R.drawable.image_splashscreen_8, "Добро есть везде и во всех", R.drawable.background_splash_screen_title_4, R.color.splash_screen_title_color_4),
        SplashScreen(R.drawable.image_splashscreen_9, "Ответственная доброта", R.drawable.background_splash_screen_title_2, R.color.splash_screen_title_color_2),
        SplashScreen(R.drawable.image_splashscreen_10, "Создание физического и психологического пространства для завершения жизни", R.drawable.background_splash_screen_title_3, R.color.splash_screen_title_color_3),
        SplashScreen(R.drawable.image_splashscreen_11, "Творческий и осознанный подход к жизни пациента", R.drawable.background_splash_screen_title_6, R.color.splash_screen_title_color_6),
        SplashScreen(R.drawable.image_splashscreen_12, "Чем больше мы принимаем добра, тем больше отдаем", R.drawable.background_splash_screen_title_1, R.color.splash_screen_title_color_1),
        SplashScreen(R.drawable.image_splashscreen_13, "Хоспис – это воплощенная гуманность", R.drawable.background_splash_screen_title_5, R.color.splash_screen_title_color_5),
        SplashScreen(R.drawable.image_splashscreen_14, "Хоспис — это призвание и служение человечеству", R.drawable.background_splash_screen_title_6, R.color.splash_screen_title_color_6),
        SplashScreen(R.drawable.image_splashscreen_15, "Хоспис – это наука помощи и искусство ухода", R.drawable.background_splash_screen_title_1, R.color.splash_screen_title_color_1),
        SplashScreen(R.drawable.image_splashscreen_16, "Ответственно и осознанно нести добро людям", R.drawable.background_splash_screen_title_3, R.color.splash_screen_title_color_3),
        SplashScreen(R.drawable.image_splashscreen_17, "Хоспис – это компетентная помощь и любовь к пациентам", R.drawable.background_splash_screen_title_4, R.color.splash_screen_title_color_4)
    ).random()
    }
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/SplashScreenActivity.kt:
```diff
@@ -0,0 +1,68 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        makeFullScreen()
```
В `onDestroy` обратно "разфулскринивать" не нужно?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/SplashScreenActivity.kt:
```diff
@@ -0,0 +1,68 @@
+@Suppress("DEPRECATION")
+class SplashScreenActivity : AppCompatActivity() {
+    private val splashscreenImages = listOf(
+        SplashScreen(R.drawable.image_splashscreen_1,"Любая помощь важна и нужна", R.drawable.background_splash_screen_title_1, R.color.splash_screen_title_color_1),
```
Всем ресурсных строк за мой счёт!

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsListFragment.kt:
```diff
@@ -0,0 +1,145 @@
+//                    binding.containerListNewsInclude.newsListRecyclerView.scrollToPosition(
+//                        0
+//                    )
+//                }
```
Удалить закомментированный код

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsListFragment.kt:
```diff
@@ -0,0 +1,145 @@
+                delay(200)
+                binding.containerListNewsInclude.newsListRecyclerView.scrollToPosition(
+                    0
+                )
```
Можно без задержки попробовать? Мне казалось, что этот код работал
```js
binding.containerListNewsInclude.newsListRecyclerView.post {
    binding.containerListNewsInclude.newsListRecyclerView.scrollToPosition(
        0
    )
}
```
Или вот ещё, как по мне, интересный вариант для проверки на работоспособность
```js
adapter.submitList(newList) {
    binding.containerListNewsInclude.newsListRecyclerView.scrollToPosition(
        0
    )
}
```

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsListFragment.kt:
```diff
@@ -0,0 +1,145 @@
+            binding.containerCustomAppBarIncludeOnFragmentNewsList.mainMenuImageButton
+        )
+        mainMenu.inflate(R.menu.menu_main)
+        mainMenu.menu.removeItem(R.id.menu_item_news)
```
Зачем удалять элемент из меню? Может нам тогда не подходит это меню, если приходится удалять из него пункты?

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsControlPanelFragment.kt:
```diff
@@ -0,0 +1,168 @@
+                    .setPositiveButton(R.string.fragment_positive_button) { dialog, _ ->
+                        dialog.cancel()
+                    }
+                    .create()
```
Диалог можно лениво создать один раз

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsControlPanelFragment.kt:
```diff
@@ -0,0 +1,168 @@
+            }
+        }
+
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
```
Можно в `onCreate` перетащить - здесь не нужно ничего от `view`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsControlPanelFragment.kt:
```diff
@@ -0,0 +1,168 @@
+            }
+        }
+
+        val activity = activity ?: return
```
`activity` в `onViewCreated` уже должна быть. используй `requireActivity()`

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/NewsControlPanelFragment.kt:
```diff
@@ -0,0 +1,168 @@
+
+            addNewsImageView.setOnClickListener {
+                findNavController().navigate(
+                    R.id.action_newsControlPanelFragment_to_createEditNewsFragment
```
Без идентификаторов, пожалуйста. Надёжнее заменить на сгенерированные классы. Если у этого направления навигации появятся аргументы, то сгенерированный `Direction` тебя обругает за непереданные аргументы, а `navigate` с идентификатором - нет, просто свалится приложение в runtime

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/CreateEditNewsFragment.kt:
```diff
@@ -0,0 +1,249 @@
+        }
+    }
+
+    private fun showErrorToast(text: Int) {
+        Toast.makeText(
+            requireContext(),
+            text,
+            Toast.LENGTH_LONG
+        ).show()
+    }
```
Вынеси наружу из этого класса и сделай глобальным расширением для `Fragment`

maks1596 replied:
Только имя у него на самом деле просто `showToast` - к ошибке данный метод не причастен

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/FilterNewsListFragment.kt:
```diff
@@ -0,0 +1,155 @@
+            }
+        }
+
+        val calendar = Calendar.getInstance()
```
Из-за того, что используется один и тот же инстанс `Calendar` как в качестве агрегатора результата выбора `DatePicker`, так и для его первичной установки, можно достичь интересного результата. Например:

№ | Действие | Результат 
--- | --- | ---
1 | Нажать на дату начала | Видим сегодняшнюю дату
2 | Выбрать "вчера" | Вчерашняя дата отобразилась в поле "Дата начала"
3 | Нажать на дату окончания | Видим вчерашнюю дату. Настораживает, но не критично
4 | Выбрать "завтра" | Завтрашняя дата отобразилась в поле "Дата окончания"
5 | Нажать на дату начала | Видим завтрашнюю дату

Ожидаемый результат:
> Видим вчерашнюю дату, потому что она отображается в нажатом поле "Дата начала"

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/MainFragment.kt:
```diff
@@ -0,0 +1,178 @@
+        mainMenu.setOnMenuItemClickListener {
+            when (it.itemId) {
+                R.id.menu_item_claims -> {
+                    findNavController().navigate(R.id.action_mainFragment_to_claimListFragment)
```
Заменить идентификаторы на сгенерированные классы `Directions` - надёжнее будет

maks1596 commented on app/src/main/java/ru/netology/fmhandroid/ui/MainFragment.kt:
```diff
@@ -0,0 +1,178 @@
+
+            expandMaterialButton.setOnClickListener {
+                when (allClaimsTextView.visibility) {
+                    View.GONE -> {
```
1. Я бы лучше наоборот ориентировался на `View.VISIBLE`, потому что есть ещё сродний `View.GONE` вариант - `View.INVISIBLE`
2. Этот `when` лучше заменить на `if else`. От `when` тут пользы никакой.
3. Лучше не ориентироваться на видимость другой `view` в таких случаях. Правильнее будет ориентироваться на текущее состояние экрана, которое хранится во `viewModel`

Dolgoff62 committed:
Исправлены зависимости для Api

Dolgoff62 committed:
добавлен fragment_auth.xml

nanstas committed:
добавил AuthFragment.kt, и поправил нейминг в вестке

Dolgoff62 committed:
Работа с авторизацией + мелкие доработки

nanstas committed:
добавлена viewModel и запросы на сохранение и обнавление токенов авторизации

Dolgoff62 committed:
Добавлены методы скрытия и показа системного status bar

nanstas committed:
замена активити на фрагмент SplashScreen + навигация

Dolgoff62 committed:
изменен метод скрытия и показа system bar

nanstas committed:
запрет обратного перехода на SplashScreenFragment по нажатию на системную кнопку

nanstas committed:
Merge remote-tracking branch 'i-teco/authentification_autorization' into authentification_autorization

Dolgoff62 committed:
Добавлен функционал по авторизации и деавторизайии юзера.

Dolgoff62 committed:
глобальный рефакторинг в нескольких классах

nanstas committed:
замена захардкоженного юзера на залогиненного в Claims

Dolgoff62 committed:
Исправлено множество ошибок и несоответствий, приводивших к "падению"

Dolgoff62 committed:
currentUser и userList перенесены в userRepository

Dolgoff62 committed:
добавлен unauthorizedInterceptor и все что с ним связано

Dolgoff62 committed:
рефактор после переноса текущего юзера в userRepository

nanstas committed:
исправлен метод сохранения созданной либо измененной заявки

nanstas committed:
исправление реализации адаптера

Dolgoff62 committed:
исправлен апи запрос на обновление токенов

Dolgoff62 committed:
исправлены фрагменты с новостями

Везде добавлен залогиненный юзер вместо "хардкода".

nanstas committed:
небольшой рефакторинг

Dolgoff62 committed:
незначительные изменения в нескольких классах

Dolgoff62 committed:
замена UnauthorizedInterceptor на RefreshAuthenticator

nanstas committed:
исправлено неккоректное поведение в RefreshAuthenticator

Dolgoff62 committed:
удалена функция расширение addAuthorizationHeader

Dolgoff62 committed:
Добавлен refreshInterceptor

Для корректной работы запроса на обновление токенов

Dolgoff62 committed:
работа над обработкой 403 ошибки

nanstas committed:
работа над обработкой 403 ошибки через StateFlow

Dolgoff62 committed:
работа над обработкой 403 ошибки через StateFlow

Перенос навигации в AppActivity

Dolgoff62 committed:
исправление метода обработки ошибки обновления токенов.

Dolgoff62 committed:
попытка обработать ошибку просроченных токенов

Dolgoff62 committed:
решена проблема с обработкой "протухшего" refresh token

Dolgoff62 committed:
Избавил проект от TODO'шек

"Лишний код под нож!!!"

Dolgoff62 committed:
удалены неиспользуемые импорты

Dolgoff62 committed:
bug fix

Исправлена ошибка при которой приложение падало на устройствах с разрешением экрана 1920х1080 из-за некорректной отрисовки фона на некоторых экранах.

Dolgoff62 committed:
bug fix

Исправлена ошибка, при которой происходила параллельная загрузка списка заявок и списка новостей при переходе на главный экран. Так же изменен подход к обновлению списков новостей и заявок на соответствующих фрагментах.

Dolgoff62 committed:
deep refactoring

Dolgoff62 committed:
изменен метод на на flatMapLatest

nanstas committed:
исправлен формат отображения короткого ФИО

nanstas committed:
добавлен разлогин во всех фрагментах

Dolgoff62 committed:
bug fix

Исправлено некорректное поведение при истекшем  refresh токене.

Dolgoff62 committed:
Добавлена проверка на права доступа к управлению новостями

Dolgoff62 committed:
bug fix

Исправлена бага с некорректным отображением имени пользователя оставившего комментарий при сбросе заявки

nanstas committed:
bug fix (отображение исполнителя в заявке после сохранения при редактирование)

Dolgoff62 committed:
переход на https

Dolgoff62 committed:
bug fix

Добавлен сертификат

Dolgoff62 committed:
bug fix

Добавлен сертификат, Исправлено отображение элементов при пустом списке новостей.

Dolgoff62 committed:
bug fix

несколько изменений

Dolgoff62 committed:
Merge remote-tracking branch 'origin/authorizatio_in_prod' into authorizatio_in_prod

Dolgoff62 committed:
обработка ошибки загрузки списка юзеров

Dolgoff62 committed:
bug fix

Исправлен баг при котором после перезапуска приложения список исполнителей при редактировании заявки был пустым

nanstas committed:
попытка обрабатывать удаление записей при рефреше новостей

nanstas committed:
bug fix
актуализация списка новостей в локальной БД

Dolgoff62 committed:
Скорректирована работа универсальной функции makeRequest

Добавлена обработка нескольких ключевых Exception

Dolgoff62 committed:
Скорректирована работа универсальной функции makeRequest - часть вторая

Dolgoff62 committed:
Убраны лишние импорты и добавлен комментарий в AuthRepository

Dolgoff62 committed:
Убран оператор "hold my beer"

Dolgoff62 committed:
Удалена одна из картинок для загрузочного экрана

Dolgoff62 committed:
исправлен номер версии сборки

Dolgoff62 committed:
Добавлен фрагмент About (О приложении)

Dolgoff62 committed:
Изменена верстка фрагмента About

Dolgoff62 committed:
добавлена навигация на aboutFragment с фрагментов со списками

nanstas committed:
добавлен разделитель между новостями

Dolgoff62 committed:
Изменен topAppBar на фрагменте about

Dolgoff62 committed:
Merge remote-tracking branch 'origin/authorizatio_in_prod' into authorizatio_in_prod

Dolgoff62 committed:
незначительные изменения

Изменен applicationId и связанная с ним директория.

Dolgoff62 committed:
Изменения в из предыдущего коммита + оптимизация импортов в классах

Dolgoff62 committed:
Изменение top app bar

А так же связанных с этим ошибок.

Dolgoff62 committed:
first try

nanstas committed:
решена проблема с запросами по HTTPS на API меньше 24

Dolgoff62 committed:
переименован trade mark image для топ апп бар

Dolgoff62 committed:
почистили проект от лишних элементов

Dolgoff62 committed:
Merge branch 'try_to_fix_certificate_error' into authorizatio_in_prod

Dolgoff62 committed:
исправлены несколько строк в ресурсах strings.xml

Dolgoff62 committed:
Merge pull request #2 from fmh-charity/authorizatio_in_prod

решена проблема с обработкой "протухшего" refresh token

Dolgoff62 committed:
Merge branch 'authentification_autorization' into road_to_first_release

nanstas committed:
актуализированы ссылки на Политику конфиденциальности и Пользовательское соглашение

Dolgoff62 committed:
добавлен обработчик нажатия системной кнопки "назад"

Dolgoff62 committed:
исправлен баг с нажатием на системную кнопку "назад"

Dolgoff62 committed:
добавлены google сервисы

Dolgoff62 committed:
обновили версию сборки, добавлены элементы в gitignore

nanstas committed:
актуализация списка заявок в локальной БД при удалении на сервере

Dolgoff62 committed:
чистка кода (лишний код под нож)

Dolgoff62 committed:
изменение version name

nanstas committed:
добавлено скрытие empty group при непустом списке заявок после фильтрации

Dolgoff62 committed:
повышен номер версии

nanstas committed:
заменен цвет TextInputLayout у поля login

Dolgoff62 committed:
Обновлены версии нескольких библиотек

nanstas committed:
заменен цвет TextInputLayout у поля password

nanstas committed:
задана минимальная дата для ввода в date pickers

nanstas committed:
задано макс. количество символов (50) для поля "Тема" в заявках

nanstas committed:
изменена логика и положение switcher

Dolgoff62 committed:
Добавлен счетчик символов для темы при создании заявки

Dolgoff62 committed:
баг фикс с дублирующимися комментариями

Dolgoff62 committed:
начало доработки фильтрации новостей

nanstas committed:
изменена верстка экранов фильтров

nanstas committed:
изменена логика фильтрации новостей

nanstas committed:
изменено отображение PopupMenu

Dolgoff62 committed:
обновлена версия сборки

Dolgoff62 committed:
Исправлен баг с бекграундом приложения

Dolgoff62 committed:
Исправлен баг с бекграундом приложения

Так же найден и исправлен баг с неотображаемыми тенями

Dolgoff62 committed:
Merge remote-tracking branch 'origin/road_to_first_release' into road_to_first_release

Dolgoff62 committed:
удалено лишнее из buidl gradle

nanstas committed:
удалены паддинги в списках

nanstas committed:
добавлен отступ и элевация у item News

Dolgoff62 committed:
добавлено уведомление об обязательных полях не заполненных юзером

При создании заявки

Dolgoff62 committed:
добавлено уведомление об обязательных полях не заполненных юзером

При создании заявки, часть вторая

Dolgoff62 committed:
убран ssl сертификат и изменено название приложения

Dolgoff62 committed:
изменен номер версии приложения

Dolgoff62 committed:
Изменено главное меню

Теперь пункт меню с фрагментом, в котором находится пользователь не удаляется, а отображается как disabled, то есть не доступен для клика

Dolgoff62 committed:
На загрузочные экраны добавлен Индикатор загрузки

Dolgoff62 committed:
обновлена версия сборки

Dolgoff62 committed:
внедрение раздела "Наша миссия"

Добавлен экран со списком элементов. Начата работа по созданию карточки "наша миссия" для ресайклера.

Dolgoff62 committed:
внедрение раздела "Наша миссия" ч.2

полностью завершена верстка элементов раздела, прописана логика, написан адаптер. Необходимо протестировать.

Dolgoff62 committed:
внедрение раздела "Наша миссия" ч.3

Закончена работа над данным разделом. Прописана вся необходимая навигация.

Dolgoff62 committed:
Добавлено главное меню во фрагмент карточки заявки

Так же добавлена анимация нажатия на кнопки в custom top app bar.

Dolgoff62 committed:
Обновлена версия билда

Dolgoff62 committed:
Изменен Base URL

Dolgoff62 committed:
Изменена иконка пустых полей при создании новости

Dolgoff62 committed:
Обновлена версия сборки

nanstas committed:
добавил триммирование полей при создании новостей, заявок и комменатариев

nanstas committed:
исправил отображение элемента на control panel

Dolgoff62 committed:
подключен proguard для релизной сборки.

Так же исправлена ошибка в authInterceptor приводившая к крашу после обфускации на моменте попытки логина в приложении.

Dolgoff62 committed:
поправлен бекграунд приложения

Dolgoff62 committed:
Заменен бэкграунд приложения
